apps/services/user-profile/src/app/v2/test/user-profile.spec.ts

```diff
@@ -0,0 +1,132 @@
+import request from 'supertest'
+
+import { createCurrentUser } from '@island.is/testing/fixtures'
+import { UserProfileScope } from '@island.is/auth/scopes'
+import { setupApp, setupAppWithoutAuth } from '@island.is/testing/nest'
+
+import { AppModule } from '../../app.module'
+import { SequelizeConfigService } from '../../sequelizeConfig.service'
+import { FixtureFactory } from './fixtureFactory'
+
+const testNationalId = '0101302399'
+
+const testUserProfile = {
+  nationalId: testNationalId,
+  email: 'test@test.is',
+  mobilePhoneNumber: '1234567',
+}
+
+describe('UserProfileController', () => {
+  describe('No auth', () => {
+    it('GET /v2/users/.national-id should return 401 when user is not authenticated', async () => {
+      // Arrange
+      const app = await setupAppWithoutAuth({
+        AppModule: AppModule,
+        SequelizeConfigService: SequelizeConfigService,
+      })
+
+      const server = request(app.getHttpServer())
+
+      // Act
+      const res = await server
+        .get('/v2/users/.national-id')
+        .set('X-Param-National-Id', '1234567890')
```

Comment:
We should just use the `testUserProfile.nationalId` as we are testing that the access token is not provided for authentication. Same in the 403 test case below.
```suggestion
        .set('X-Param-National-Id', testUserProfile.nationalId)
```